### PR TITLE
gpsd: move libgpsdpacket.so under -devel subpackage

### PR DESCRIPTION
**Pull request by JamiKettunen**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: common/shlibs [hunk 1]
```diff
@@ -4035,3 +4035,4 @@ libstemmer.so.2 libstemmer-2.1.0_1
 libdecor-0.so.0 libdecor-0.1.0_1
 libavif.so.13 libavif-0.9.3_1
 libkdumpfile.so.8 libkdumpfile-0.4.0_1
+libgpsdpacket.so.29 gpsd-3.23.1_3
```
Comment (ericonr) [inline common/shlibs:4038]: This warning is harmless, so please leave it out for now. It's necessary only if some other package depends on it.

Comment (ericonr) [reply common/shlibs]: Moving the .so is fine :)

If you want, `vmove "usr/lib/*.so"` should also work.

File: srcpkgs/gpsd/template [hunk 2]
```diff
@@ -105,6 +105,7 @@ gpsd-devel_package() {
 		vmove usr/include
 		vmove "usr/lib/pkgconfig/libgps*.pc"
 		vmove usr/lib/libgps.so
```
Comment (abenson) [inline srcpkgs/gpsd/template:107]: Could these be combined to be just `vmove "usr/lib/*.so"`?

Comment (abenson) [reply srcpkgs/gpsd/template:107]: This whole section could be changed to

    vmove usr/include
    vmove usr/lib/pkgconfig
    vmove "usr/lib/*.so"
    vmove usr/share/man/man3

Comment (Piraty) [reply srcpkgs/gpsd/template:107]: in such scenarios, implicit subpackages (like alpine does) seem appealing

Comment (abenson) [reply srcpkgs/gpsd/template:107]: Ah, I didn't think about the other `-devel` package.  This is fine as is

**Pull request merged**